### PR TITLE
feat(dogfood): add `MsgUpdateParams`

### PR DESCRIPTION
**Pull request by MaxMustermann2**
This change allows the live editing of dogfood params by sending transactions. Such transactions must be sent by the authority, which is currently set to the governance-controlled address. If the transaction does not contain all of the required parameters, they are loaded from the previously set parameters. If the sent transaction has invalid parameters (whether stateful or stateless), they are again overwritten from the previous set.

In addition, this PR removes the usage of the deprecated `x/params` module from the `x/dogfood` keeper.

## Summary by CodeRabbit

- **New Features**
  - Introduced the `UpdateParams` command in the CLI for updating module parameters.
  - Added `MsgUpdateParams` and `MsgUpdateParamsResponse` messages for enhanced communication.
  - Updated the `Msg` service with the new `UpdateParams` method for better parameter management.

- **Improvements**
  - Enhanced parameter validation and logging in the `UpdateParams` function to ensure better error handling and traceability.

### CODE REVIEW

**Comment by MaxMustermann2** [comment]
~The build failure is because `gitleaks` is treating the `exo....` address of `operator2` as an API key. I tried to add it to ignore, but this approach didn't work. I will try it on my local via [act](https://github.com/nektos/act) till I find the solution.~

**Comment by adu-web3** [comment]
this could be used for testing but seems not very good to use a hardcoded address as authorized transaction sender in production environment.

the typical approach to update module params is through governance proposals(customize proposal message in module and implement govern module hooks). While it seems difficult implementing this before we launch the mainnet, I think a hardcoded authority is not a good pattern

**Comment by MaxMustermann2** [comment]
> this could be used for testing but seems not very good to use a hardcoded address as authorized transaction sender in production environment.
> 
> the typical approach to update module params is through governance proposals(customize proposal message in module and implement govern module hooks). While it seems difficult implementing this before we launch the mainnet, I think a hardcoded authority is not a good pattern

Maybe we can parameterize this as a CLI argument - if unset, it defaults to the governance module as the authority; otherwise, it starts the node with the supplied address as the authority. For testnet, we can supply an EOA as the authority, while mainnet can continue with the module by default.

**Comment by MaxMustermann2** [comment]
> Maybe we can parameterize this as a CLI argument - if unset, it defaults to the governance module as the authority; otherwise, it starts the node with the supplied address as the authority. For testnet, we can supply an EOA as the authority, while mainnet can continue with the module by default.

This will be done in a separate PR. For now, this PR just adds functionality to edit the params by using governance.

**Comment by MaxMustermann2** [comment]
@coderabbitai review

File: x/dogfood/keeper/abci.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ func (k Keeper) EndBlock(ctx sdk.Context) []abci.ValidatorUpdate {
 		prevMap[addressString] = validator.Power
 	}
 	operators, keys := k.operatorKeeper.GetActiveOperatorsForChainID(ctx, ctx.ChainID())
+	// TODO: get only assetIDs and change to vote power call
```
[the remark this answers is not in the document]
Comment (MaxMustermann2) [reply x/dogfood/keeper/abci.go:79]: Sure, please create one with reference to the `x/dogfood` module.

File: x/dogfood/keeper/unbonding.go [hunk 2]
```diff
@@ -11,15 +11,15 @@ import (
...
 	epochInfo, _ := k.epochsKeeper.GetEpochInfo(
-		ctx, k.GetEpochIdentifier(ctx),
+		ctx, params.EpochIdentifier,
 	)
 	// if i execute the transaction at epoch 5, the vote power change
 	// goes into effect at the beginning of epoch 6. the information
 	// should be held for 7 epochs, so it should be deleted at the
 	// beginning of epoch 13 or the end of epoch 12.
-	return epochInfo.CurrentEpoch + int64(unbondingEpochs) // #nosec G701
+	return epochInfo.CurrentEpoch + int64(params.EpochsUntilUnbonded) // #nosec G701
```
[the remark this answers is not in the document]
Comment (MaxMustermann2) [reply x/dogfood/keeper/unbonding.go:22]: This is handled at genesis. Not required.

**Review by leonz789 — APPROVED** [review]

**Comment by MaxMustermann2** [comment]
Included in #125

**Pull request closed without merging**